test/api/requests/user-action.test.js

```diff
@@ -0,0 +1,95 @@
+const { expect } = require('chai');
+const app = require('../../../app');
+const request = require('supertest');
+const { authenticatedSession } = require('../support/session');
+const validateAgainstJSONSchema = require('../support/validateAgainstJSONSchema');
+const factory = require('../support/factory');
+const userActionFactory = require('../support/factory/user-action');
+
+const version = '/v0';
+const resource = 'site';
+const route = 'user-action';
+
+const notAuthenticatedError = 'You are not permitted to perform this action. Are you sure you are logged in?';
+const notAuthorizedError = 'You are not authorized to perform that action';
+
+const path = id => `${version}/${resource}/${id}/${route}`;
+
+const makeGetRequest = (expectedCode, { id, cookie = null }) => {
```

Comment:
nice, we should use helpers like this for other similar tests

Reply:
yeah, I think we could write a small helper library like we use for our wrapper around the `fetch` api, and return different functions for each of the HTTP verbs that we use